stackit/internal/services/iaas/image/resource.go

```diff
@@ -0,0 +1,851 @@
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *imageResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve values from state
+	var model Model
+	diags := req.State.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectId := model.ProjectId.ValueString()
+	imageId := model.ImageId.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "image_id", imageId)
+
+	// Delete existing image
+	err := r.client.DeleteImage(ctx, projectId, imageId).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting image", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+	_, err = wait.DeleteImageWaitHandler(ctx, r.client, projectId, imageId).WaitWithContext(ctx)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting image", fmt.Sprintf("image deletion waiting: %v", err))
+		return
+	}
+
+	tflog.Info(ctx, "Image deleted")
+}
+
+// ImportState imports a resource into the Terraform state on success.
+// The expected format of the resource import identifier is: project_id,image_id
+func (r *imageResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, core.Separator)
+
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing image",
+			fmt.Sprintf("Expected import identifier with format: [project_id],[image_id]  Got: %q", req.ID),
+		)
+		return
+	}
+
+	projectId := idParts[0]
+	imageId := idParts[1]
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "image_id", imageId)
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), projectId)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("image_id"), imageId)...)
+	tflog.Info(ctx, "Image state imported")
+}
+
+func mapFields(ctx context.Context, imageResp *iaasalpha.Image, model *Model) error {
+	if imageResp == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+
+	var imageId string
+	if model.ImageId.ValueString() != "" {
+		imageId = model.ImageId.ValueString()
+	} else if imageResp.Id != nil {
+		imageId = *imageResp.Id
+	} else {
+		return fmt.Errorf("image id not present")
+	}
+
+	idParts := []string{
+		model.ProjectId.ValueString(),
+		imageId,
+	}
+	model.Id = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+
+	// Map config
+	var configModel = &configModel{}
+	var configObject basetypes.ObjectValue
+	diags := diag.Diagnostics{}
+	if imageResp.Config != nil {
+		configModel.BootMenu = types.BoolPointerValue(imageResp.Config.BootMenu)
```

Comment:
maybe extract this part to a `mapConfig` method

Reply:
Yeah it's a bit more lengthy than the other two objects, but I think it is simpler to have all of them in the same function and tested together